src/int/div.rs

```diff
@@ -0,0 +1,335 @@
+//! [`Int`] division operations.
+
+use core::ops::Div;
+
+use subtle::{Choice, CtOption};
+
+use crate::{CheckedDiv, ConstChoice, ConstCtOption, Int, NonZero, Uint};
+
+impl<const LIMBS: usize> Int<LIMBS> {
+    #[inline]
+    /// Base div_rem operation.
+    /// Given `(a, b)`, computes the quotient and remainder of their absolute values. Furthermore,
+    /// returns the signs of `a` and `b`.
+    const fn div_rem_base(
+        &self,
+        rhs: &NonZero<Self>,
+    ) -> (Uint<{ LIMBS }>, Uint<{ LIMBS }>, ConstChoice, ConstChoice) {
+        // Step 1: split operands into signs and magnitudes.
+        let (lhs_mag, lhs_sgn) = self.abs_sign();
+        let (rhs_mag, rhs_sgn) = rhs.0.abs_sign();
+
+        // Step 2. Divide magnitudes
+        // safe to unwrap since rhs is NonZero.
+        let (quotient, remainder) = lhs_mag.div_rem(&NonZero::<Uint<LIMBS>>::new_unwrap(rhs_mag));
```

Comment:
Perhaps you could define `abs_sign` on `NonZero<Int<LIMBS>>` and have it return `NonZero`?

Reply:
Done.